src/qilib/utils/serialization.py

```diff
@@ -20,18 +20,68 @@
 import base64
 from functools import partial
 from json import JSONDecoder, JSONEncoder
-from typing import Any, Callable, Dict, Tuple, Optional
+from typing import Any, Callable, Dict, Tuple, Optional, Union
 from dataclasses_json.api import DataClassJsonMixin
-
 import numpy as np
 
-from qilib.data_set.mongo_data_set_io import MongoDataSetIO
+from qilib.utils.type_aliases import EncodedNumpyArray, numpy_ndarray_type, NumpyKeys
+
 
 # A callable type for transforming a given argument with a type to another type
 TransformFunctionResult = Any
 TransformFunction = Callable[[Any], TransformFunctionResult]
 
 
+class NumpyArrayEncDec:
```

Comment:
For my understanding: what is the reason to create a class with only static methods in Python, when we can also create a module exposing the methods directly?

Reply:
For this change the reason was only packaging methods that belong together instead of putting them in this file as a bunch public methods (which they were) which was inconsistent with the JSON encoder/decoder in this same file which is a separate class.
However, your suggestion is valid but then I would propose to not only do that for this numpy array encoder/decoder but to create a separate package for all encoders/decoders in qilib (numpy array, json encoder/decoder and the different encoding/decoding methods in serializer). This is beyond the scope of this ticket but I am glad to add a ticket to our backlog if we need to do this.